js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode2025.tsx

```diff
@@ -0,0 +1,238 @@
+              <StaleReasonsTag liveData={liveData} assetKey={definition.assetKey} />
+              <ChangedReasonsTag
+                changedReasons={definition.changedReasons}
+                assetKey={definition.assetKey}
+              />
+            </Box>
+          ) : (
+            <div style={{minHeight: HOVER_EXPAND_MARGIN}} />
+          )}
+          <AssetNodeBox $selected={selected} $isMaterializable={definition.isMaterializable}>
+            <AssetNameRow definition={definition} />
+            {facets.has(AssetNodeFacet.Description) && (
+              <AssetNodeRow label={null}>
+                {definition.description ? (
+                  <AssetDescription $color={Colors.textDefault()}>
+                    {markdownToPlaintext(definition.description).split('\n')[0]}
+                  </AssetDescription>
+                ) : (
+                  <AssetDescription $color={Colors.textDefault()}>No description</AssetDescription>
+                )}
+              </AssetNodeRow>
+            )}
+            {facets.has(AssetNodeFacet.Owner) && (
+              <AssetNodeRow label={labelForFacet(AssetNodeFacet.Owner)}>
+                {definition.owners.length > 0 ? (
+                  <SingleOwnerOrTooltip owners={definition.owners} />
+                ) : null}
+              </AssetNodeRow>
+            )}
+            {facets.has(AssetNodeFacet.LatestEvent) && (
+              <AssetNodeRow label={labelForFacet(AssetNodeFacet.LatestEvent)}>
+                {assetNodeLatestEventContent({definition, liveData})}
+              </AssetNodeRow>
+            )}
+            {facets.has(AssetNodeFacet.Checks) && (
+              <AssetNodeRow label={labelForFacet(AssetNodeFacet.Checks)}>
+                {liveData && liveData.assetChecks.length > 0 ? (
+                  <Link
+                    to={assetDetailsPathForKey(definition.assetKey, {view: 'checks'})}
+                    onClick={(e) => e.stopPropagation()}
+                  >
+                    <AssetChecksStatusSummary
+                      liveData={liveData}
+                      rendering="dag2025"
+                      assetKey={definition.assetKey}
+                    />
+                  </Link>
+                ) : null}
+              </AssetNodeRow>
+            )}
+            {facets.has(AssetNodeFacet.Freshness) && (
+              <AssetNodeRow label={labelForFacet(AssetNodeFacet.Freshness)}>
+                {!liveData?.freshnessInfo ? null : isAssetOverdue(liveData) ? (
+                  <Box flex={{gap: 4, alignItems: 'center'}}>
+                    <Icon name="close" color={Colors.accentRed()} />
+                    Violated
+                  </Box>
+                ) : (
+                  <Box flex={{gap: 4, alignItems: 'center'}}>
+                    <Icon name="done" color={Colors.accentGreen()} />
+                    Passing
+                  </Box>
+                )}
+              </AssetNodeRow>
+            )}
```

Comment:
I think here we might want to conditionally use our new concept of freshness.

Reply:
https://github.com/dagster-io/dagster/pull/29319

Reply:
Yeah this is currently the "old" freshness that is currently shown in the asset graph sidebar as well. Will add this to the list of things to do in a follow-up today!